pinecone/manage.py

```diff
@@ -194,14 +190,14 @@ def describe_index(name: str):
     """
     api_instance = _get_api_instance()
     response = api_instance.describe_index(name)
-    db = response['database']
-    ready = response['status']['ready']
-    state = response['status']['state']
-    return IndexDescription(name=db['name'], metric=db['metric'],
-                            replicas=db['replicas'], dimension=db['dimension'], shards=db['shards'],
-                            pods=db.get('pods', db['shards'] * db['replicas']), pod_type=db.get('pod_type', 'p1'),
-                            status={'ready': ready, 'state': state}, metadata_config=db.get('metadata_config'),
-                            source_collection=db.get('source_collection', ''))
+    db_status = response.get('status', {})
+    response = response['database'].to_dict()
+    state = db_status.get('state', '')
```

Comment:
Hmmm okay, i was thinking about what would be a good thing to do when the fields are deprecated in the future. Maybe we should stick to existing data types rather than None. just thinking out loud